qiskit/assembler/assemble_circuits.py

```diff
@@ -402,7 +393,54 @@ def assemble_circuits(
 
     if calibrations and calibrations.gates:
         qobj_config.calibrations = calibrations
+    with warnings.catch_warnings():
+        # The class QasmQobj is deprecated
+        warnings.filterwarnings("ignore", category=DeprecationWarning, module="qiskit")
+        return QasmQobj(
+            qobj_id=qobj_id, config=qobj_config, experiments=experiments, header=qobj_header
+        )
 
-    return QasmQobj(
-        qobj_id=qobj_id, config=qobj_config, experiments=experiments, header=qobj_header
-    )
+
+@deprecate_func(
+    since="1.2",
+    removal_timeline="in the 2.0 release",
+    additional_msg="The function assemble_circuits is being deprecated "
+    "as they are not necessary for BackendV2. If user still need Qobj, that probably "
+    "means that they are using a backend based on the deprecated BackendV1 class.",
```

Comment:
I think it's not necessary to mention the function that is being deprecated in the additional message, the decorator already introduces the name in the message, see sample output:

```
DeprecationWarning: The function ``qiskit.compiler.assembler.assemble()`` is deprecated as of qiskit 1.2. It will be removed in the 2.0 release. 
The function assemble is being deprecated as they are not necessary for BackendV2. 
If user still need Qobj, that probably means that they are using a backend based on the deprecated BackendV1 class.
```

So I suggest creating a generic additional message that you can directly copy-paste in all deprecations:

```suggestion
    additional_msg="The `Qobj` class and related functionality are part of the deprecated `BackendV1` "
    "workflow,  and no longer necessary for `BackendV2`. If a user workflow requires `Qobj` it likely relies "
    "on deprecated functionality and should be updated to use `BackendV2`.",
```

Reply:
much better. Thanks! I added this `additional_msg` in all the decorators.